cmd/bedFormat/bedFormat.go

```diff
@@ -99,7 +133,64 @@ func bedFormat(s Settings) {
 		if s.ScaleNameFloat != 1 {
 			v.Name = fmt.Sprintf("%.8g", s.ScaleNameFloat*parse.StringToFloat64(v.Name))
 		}
-		bed.WriteBed(out, v)
+
+		if s.FdrAnnotation {
+			totalWindows++
+			if s.RawPValueAnnotationField >= len(v.Annotation) {
+				log.Fatalf("Error: rawPValueAnnotationField, %v, exceeds the length of the annotation slice in bed entry: %v.\n", s.RawPValueAnnotationField, len(v.Annotation))
+			}
+			if mapEntry, inMap = fdrMap[parse.StringToFloat64(v.Annotation[s.RawPValueAnnotationField])]; inMap {
+				mapEntry.Count++
+				mapEntry.RawPValue = parse.StringToFloat64(v.Annotation[s.RawPValueAnnotationField])
+				fdrMap[parse.StringToFloat64(v.Annotation[s.RawPValueAnnotationField])] = mapEntry
+			} else {
+				fdrMap[parse.StringToFloat64(v.Annotation[s.RawPValueAnnotationField])] = FdrConverter{Count: 1, Rank: 0, RawPValue: parse.StringToFloat64(v.Annotation[s.RawPValueAnnotationField]), AdjPValue: 0}
+			}
+
+			bed.WriteBed(tmp, v)
+		} else {
+			bed.WriteBed(out, v)
+		}
+	}
+
+	if s.FdrAnnotation {
+		err = tmp.Close()
+		exception.PanicOnErr(err)
+
+		var currRank int = 0
+		var fdrSlice []FdrConverter = make([]FdrConverter, 0)
+		for i := range fdrMap {
+			fdrSlice = append(fdrSlice, fdrMap[i])
+		}
+
+		sort.Slice(fdrSlice, func(i, j int) bool { return compareFdrConverter(fdrSlice[i], fdrSlice[j]) == -1 })
+
+		for i := len(fdrSlice) - 1; i >= 0; i-- {
+			currRank += fdrSlice[i].Count
```

Comment:
Does the absolute number of currRank matter in adjusting p-value?

Right now, it seems like if i=len(fdrSlice)-1, fdrSlice[i].Count=2, since we start with currRank=0, then currRank becomes currRank += fdrSlice[i].Count (0+2=2), so "the 1st highest raw p-value is now ranked 2nd because it appears 2 times". Is this intended behavior? Or do we want "the 1st highest raw p-value is ranked 1st, but the 2nd highest raw p-value will be ranked 3rd because the 1st highest raw p-value appears 2 times", in which case I think we would do fdrSlice[i].Rank = currRank before currRank += fdrSlice[i].Count, aka swapping lines 169 and 170

Reply:
(note to self: higher rank means lower p-value)